lib/rdf-borsh/src/borsh_writer.rs

```diff
@@ -39,10 +47,16 @@ impl BorshWriter {
 
     #[allow(unused_mut)]
     pub fn finish(mut self) -> Result<()> {
-        let version = 0u8;
+        let magic_number = b"RDFB";
+        self.sink.write_all(&magic_number[..])?;
+
+        let version = 0x01u8;
```

Comment:
The version number should be `'1'` (ASCII 0x31), not 0x01. Not your fault, the autogenerated spec in the RDF.rb repo doesn't match the actual implementation on this, I'll need to update it.